karapace/backup/api.py

```diff
@@ -303,15 +293,34 @@ def _write_partition(
     )
 
 
+def _handle_restore_topic_legacy(
+    instruction: RestoreTopicLegacy,
+    config: Config,
+) -> None:
+    if config["topic_name"] != instruction.name:
+        LOG.warning(
+            "Not creating topic, because the name %r from the config and the name %r from the CLI differ.",
+            config["topic_name"],
+            instruction.name,
+        )
+        return
+    _maybe_create_topic(
```

Comment:
Should we have similar option as for the backup files that overwriting is not allowed by default? The logic for creating the topic will just log `Topic already exist` and allows to proceed.

Reply:
Yes, I'm still finishing this part but that's the idea. For v1 and v2 the behaviour should not be changed (just logging), but for V3 the topic creation will not be allowed if the topic already exists